src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/ECDiffieHellmanWrapper.cs

```diff
@@ -0,0 +1,209 @@
+            ReadOnlySpan<byte> passwordBytes,
+            ReadOnlySpan<byte> source,
+            out int bytesRead) =>
+            _wrapped.ImportEncryptedPkcs8PrivateKey(passwordBytes, source, out bytesRead);
+
+        public override void ImportEncryptedPkcs8PrivateKey(
+            ReadOnlySpan<char> password,
+            ReadOnlySpan<byte> source,
+            out int bytesRead) =>
+            _wrapped.ImportEncryptedPkcs8PrivateKey(password, source, out bytesRead);
+
+        public override void ImportPkcs8PrivateKey(ReadOnlySpan<byte> source, out int bytesRead) =>
+            _wrapped.ImportPkcs8PrivateKey(source, out bytesRead);
+
+        public override void ImportSubjectPublicKeyInfo(ReadOnlySpan<byte> source, out int bytesRead) =>
+            _wrapped.ImportSubjectPublicKeyInfo(source, out bytesRead);
+
+        public override void ImportECPrivateKey(ReadOnlySpan<byte> source, out int bytesRead) =>
+            _wrapped.ImportECPrivateKey(source, out bytesRead);
+
+        public override byte[] ExportECPrivateKey() => _wrapped.ExportECPrivateKey();
+
+        public override bool TryExportECPrivateKey(Span<byte> destination, out int bytesWritten) =>
+            _wrapped.TryExportECPrivateKey(destination, out bytesWritten);
+
+        public override void ImportFromPem(ReadOnlySpan<char> input) => _wrapped.ImportFromPem(input);
+
+        public override void ImportFromEncryptedPem(ReadOnlySpan<char> input, ReadOnlySpan<char> password) =>
+            _wrapped.ImportFromEncryptedPem(input, password);
+
+        public override void ImportFromEncryptedPem(ReadOnlySpan<char> input, ReadOnlySpan<byte> passwordBytes) =>
+            _wrapped.ImportFromEncryptedPem(input, passwordBytes);
+
+        public override int KeySize
+        {
+            get => _wrapped.KeySize;
+            set => _wrapped.KeySize = value;
+        }
+
+        public override KeySizes[] LegalKeySizes => _wrapped.LegalKeySizes;
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _wrapped.Dispose();
+            }
+        }
+
+        public override byte[] ExportEncryptedPkcs8PrivateKey(
+            ReadOnlySpan<byte> passwordBytes,
+            PbeParameters pbeParameters) =>
+            _wrapped.ExportEncryptedPkcs8PrivateKey(passwordBytes, pbeParameters);
+
+        public override byte[] ExportEncryptedPkcs8PrivateKey(
+            ReadOnlySpan<char> password,
+            PbeParameters pbeParameters) =>
+            _wrapped.ExportEncryptedPkcs8PrivateKey(password, pbeParameters);
+
+        // Do not wrap ExportPkcs8PrivateKey, let it fall back to reconstructing it from parameters
+        // so that the ECDiffieHellman.Create()-returned object uses the same set of attributes on all platforms.
+        // (CNG adds the key usage attribute to distinguish ECDSA from ECDH)
+        //public override byte[] ExportPkcs8PrivateKey() => _wrapped.ExportPkcs8PrivateKey();
```

Comment:
Interesting. It this just so we get bit-for-bit (or thereabouts) the same exports on platforms? Seems reasonable, but is there now a discrepancy between this attribute's presence between encrypted and non-encrypted PKCS8 exports?

Reply:
Hm, you're right, I might need to block the encrypted ones as well.

We had a test for the unencrypted ones.  Encrypted export from CNG is special, since they have complicated export permissions... but that shouldn't apply for ECDsa.Create() keys.